src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    internal static class OrderingHelper
+    {
+        private enum MoveAction { Up=0, Down }
+
+        /// <summary>
+        /// Get project item elements based on the project tree.
+        /// Project tree can be a folder or item.
+        /// </summary>
+        private static IEnumerable<ProjectItemElement> GetElements(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree)
+        {
+            var treeQueue = new Queue<IProjectTree>();
+            var lookup = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            if (projectTree.IsFolder)
+            {
+                treeQueue.Enqueue(projectTree);
+            }
+            else
+            {
+                if (projectTree is IProjectItemTree2 projectItemTree && IsValidDisplayOrder(projectItemTree.DisplayOrder))
+                {
+                    lookup.Add(unconfiguredProject.MakeRooted(projectTree.FilePath));
+                }
+            }
+
+            while (treeQueue.Count > 0)
+            {
+                var tree = treeQueue.Dequeue();
+
+                foreach (var childTree in tree.Children)
+                {
+                    // If the child tree is a folder, add it to the queue for further processing.
+                    if (childTree.IsFolder)
+                    {
+                        treeQueue.Enqueue(childTree);
+                    }
+                    else
+                    {
+                        if (childTree is IProjectItemTree2 childItemTree && IsValidDisplayOrder(childItemTree.DisplayOrder))
+                        {
+                            lookup.Add(unconfiguredProject.MakeRooted(childTree.FilePath));
+                        }
+                    }
+                }
+            }
+
+            var elements = new List<ProjectItemElement>();
+
+            foreach (var item in project.AllEvaluatedItems)
+            {
+                if (lookup.Contains(unconfiguredProject.MakeRooted(item.EvaluatedInclude)))
+                {
+                    elements.Add(item.Xml);
+                }
+            }
+
+            return elements;
+        }
+
+        /// <summary>
+        /// Gets the display order for a project tree.
+        /// </summary>
+        private static int GetDisplayOrder(IProjectTree projectTree)
+        {
+            if (projectTree is IProjectTree2 projectTree2)
+            {
+                return projectTree2.DisplayOrder;
+            }
+            return 0;
```

Comment:
Can you explain why it is safe to return `0` here?

Reply:
I'll add a comment. It's safe to return because the display order value of nodes that are not set explicitly are always zero.